### PR TITLE
:sparkles: [#2175] added configureable zoom and start location

### PR DESCRIPTION
**Pull request by bart-maykin**
Fixes open-formulieren/open-forms#2175

- retrieved default zoom and location from configuration
- set starting location as configured latitude and longitude 
- set starting zoomlevel to be either map default or the configured zoomlevel

### CODE REVIEW

File: src/formio/components/Map.js [hunk 1]
```diff
@@ -90,16 +91,26 @@ export default class Map extends Field {
...
     if (!container) return;
 
     this.reactRoot.render(
       <LeafletMap
-        markerCoordinates={markerCoordinates || null}
+        markerCoordinates={markerCoordinates}
         onMarkerSet={this.onMarkerSet.bind(this)}
+        zoomLevel={zoom || MAP_DEFAULTS.zoom}
       />
     );
```
Comment (sergei-maertens) [inline src/formio/components/Map.js:115]: this isn't correct - the marker implies the select location and when no default value is provided, there just initially are no coordinates selected. Therefor, `markerCoordinates` can still be `null`.

It looks like you're trying to use `markerCoordinates` to set the center of the map view, but that's an entirely separate concept/prop that should be added.

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: src/components/Map/index.js [hunk 2]
```diff
@@ -49,7 +55,7 @@ const LeaftletMap = ({markerCoordinates, onMarkerSet, disabled = false}) => {
       <TileLayer url={TILE_LAYERS.url} {...TILE_LAYERS.options} />
       {coordinates ? (
         <>
-          <MapView coordinates={coordinates} />
+          <MapView coordinates={defaultCenter} zoomLevel={defaultZoomLevel} />
```
Comment (bart-maykin) [inline src/components/Map/index.js:58]: Double check if this doesn't reset the center view and zoom level when user places a marker.

File: src/formio/components/Map.js [hunk 3]
```diff
@@ -91,15 +92,23 @@ export default class Map extends Field {
   }
 
   renderReact() {
+    const defaultCenter =
+      this.component.initialCenter.lat && this.component.initialCenter.lng
+        ? [this.component.initialCenter.lat, this.component.initialCenter.lng]
+        : DEFAULT_LAT_LON;
```
Comment (bart-maykin) [inline src/formio/components/Map.js:98]: maybe

```js
const [defaultLat, defaultLon] = DEFAULT_LAT_LON;
const {lat = defaultLat, lng = defaultLon} = this.component?.initalCenter || {};
const defaultCenter = [lat, lng];
```

File: src/components/Map/Map.stories.js [hunk 4]
```diff
@@ -12,11 +12,21 @@ export default {
   decorators: [withMapLayout],
   args: {
     markerCoordinates: [52.1326332, 5.291266],
+    defaultCenter: [52.1326332, 5.291266],
```
Comment (bart-maykin) [inline src/components/Map/Map.stories.js:15]: changing this does not appear to have an effect in storybook -> check if it's also in the component itself but either way this should work as expected

**Review by sergei-maertens — APPROVED** [review]

**Pull request merged**